src/apps/actions.js

```diff
@@ -18,10 +19,23 @@ export function newError(error) {
     };
 }
 
-export function fetchBundleError(bundleId, error) {
+export function fetchBundleError(bundleId, error, bundle) {
     return {
         type: FETCH_BUNDLE_ERROR,
         bundleId,
+        bundle,
         error
     };
 }
+
+export function promulgateBundle(bundle, bundleId, juttleServiceHost, dispatch) {
+    dispatch(newBundle(bundleId, bundle, []));
+
+    return api.describe(juttleServiceHost, bundle)
+    .then(inputs => {
+        dispatch(newBundle(bundleId, bundle, inputs));
+    })
+    .catch(err => {
+        dispatch(fetchBundleError(bundleId, err, bundle));
+    });
```

Comment:
instead of doing this lets add redux's [thunk middleware](https://github.com/gaearon/redux-thunk) and write the function like this:
```javascript
function promulgateBundle(bundle, bundleId) {
    return (dispatch, getState) => {
        const { juttleServiceHost } = getState();
        // rest of your stuff
    }
}
```